test/test_regression.py

```diff
@@ -0,0 +1,312 @@
+""" Regression tests for https://github.com/nameko/nameko/issues/428
```

Comment:
not sure what you mean by "as a reference" but i guess i'd lean towards keeping. how come they are so slow?